printer/dot/printer.go

```diff
@@ -33,14 +42,55 @@ func (d Dot) Print(g *graph.Graph, w io.Writer) error {
 		if pv == nil {
 			pv = provider.RawProvider{}
 		}
-		shape := "ellipse"
+		attr := map[string]string{
+			"shape": "ellipse",
+		}
 		if pv.IsEdge(rs) {
-			shape = "rectangle"
+			attr["shape"] = "rectangle"
+		}
+
+		if opt.ShowIcons && n.Resource.Icon != "" {
+			ext := path.Ext(n.Resource.Icon)
+			pngIcon := fmt.Sprintf("%s.png", n.Resource.Icon[0:len(n.Resource.Icon)-len(ext)])
+			assetPath := path.Join("inframap", "assets", pv.Type().String(), pngIcon)
+			pathIcon := path.Join(xdg.CacheHome, assetPath)
+
+			attr["image"] = fmt.Sprintf("%q", pathIcon)
+			attr["shape"] = "plaintext"
+			attr["labelloc"] = "b"
+			attr["height"] = "1.15"
+
+			// If the file does not exists on the Cache path, we have to write it,
+			// if not it means it's already correct so nothing to be done
+			if _, err := os.Stat(pathIcon); os.IsNotExist(err) {
```

Comment:
do we have a way to invalidate the cache ? let's suppose the icons are modified `inframap` side, the final user will still use the the old icons.
it's out of context of this PR but maybe we can open an issue for this.

Reply:
Yes I thought about it, but as you mention is mostly outside of the context of this PR as it's an edge case that it's not that often.

One way would be to check the content (like a hash) and check if it changes, and if it does jut drop it all as generating anew one it's fast.